### ISSUE TITLE
Warpify subshells doesn't work at all with FISH

### ISSUE THREAD

**Issue by Soromeister** [warpdotdev/Warp#3968]
### Describe the bug

When activating `Warpify subshell`, a text is being inserted into `stdin` that can't be executed and nothing gets warpified.

### To reproduce

1. Use `zsh`
2. Type `fish`
3. Press `Warpify subshell`

### Expected behavior

Subshell gets warpified

### Screenshots

https://github.com/warpdotdev/Warp/assets/45243695/45d1f885-e576-457a-bde2-78fc49d37ccd

### Operating system

MacOS

### Operating system and version

Sonoma 14.2

### Shell Version

zsh 5.9 (arm-apple-darwin21.3.0)

### Current Warp version

v0.2023.12.05.08.02.stable_00

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Recent working Warp date

N/A

### Additional context

Using [TIDE](https://github.com/IlanCosman/tide) Theme for FISH, but this also happens when not using it.

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

N/A, the issue only applies to a Warp specific feature. (i.e. Warp AI, Workflows)

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Hi @Soromeister Thanks for submitting, Just out of curiosity, which version of fish are you using? run `fish --version` (if lower than 3.6, please update). 

Also, can you try adding the following in your fish config and see if the issue persists? 
```
if status is-interactive
  printf '\eP$f{"hook": "SourcedRcFileForWarp", "value": { "shell": "fish"}}\x9c'
end
```

To anyone else facing this issue, please add a :+1: to the original post at the top or comment with your details, and subscribe if you'd like to be notified.

**Comment by dannyneira** [comment]
@Soromeister any luck with the above?

**Comment by Soromeister** [comment]
Hi, my fish version was `3.6.x`. I've upgraded to `3.7.0` and with that `printf` line added in the fish config file it will warpify it, but it completely breaks the prompt. This seems to be related to some plugins I'm using (I already removed the TIDE prompt). Removing the `~/.config/fish` folder and executing fish will warpify it too. I think it's the version `3.7.0` which actually fixed it on a vanilla install.
